llvm/test/CodeGen/RISCV/double-imm.ll

```diff
@@ -17,8 +17,8 @@ define double @double_imm() nounwind {
 ;
 ; CHECK64D-LABEL: double_imm:
 ; CHECK64D:       # %bb.0:
-; CHECK64D-NEXT:    lui a0, %hi(.LCPI0_0)
-; CHECK64D-NEXT:    fld fa0, %lo(.LCPI0_0)(a0)
+; CHECK64D-NEXT:    lui a0, %hi(.promoted_doubles.double_imm)
```

Comment:
Hm, looks like maybe we shouldn't apply the transform unless there's more than one double in the function?

Reply:
Agreed, added.